dominion/src/dominion/player.rs

```diff
@@ -85,44 +86,44 @@ mod tests {
     fn test_draw_card_no_shuffle() {
         let mut p = Player::new();
 
-        p.deck_pile.clear();
-        assert!(p.deck_pile.is_empty());
+        p.draw_pile.clear();
+        assert!(p.draw_pile.is_empty());
 
-        p.deck_pile.push(CardKind::Copper);
-        p.deck_pile.push(CardKind::Silver);
+        p.draw_pile.push(CardKind::Silver);
+        p.draw_pile.push(CardKind::Copper);
 
         p.draw_card();
 
-        assert_eq!(p.deck_pile, vec![CardKind::Silver]);
+        assert_eq!(p.draw_pile, vec![CardKind::Silver]);
         assert_eq!(p.hand, vec![CardKind::Copper]);
     }
 
     #[test]
     fn test_draw_card_shuffle() {
         let mut p = Player::new();
 
-        p.deck_pile.clear();
-        assert!(p.deck_pile.is_empty());
+        p.draw_pile.clear();
+        assert!(p.draw_pile.is_empty());
 
         for _ in 0..5 {
             p.discard_pile.push(CardKind::Copper);
         }
 
         p.draw_card();
 
-        assert_eq!(p.deck_pile, vec![CardKind::Copper; 4]);
+        assert_eq!(p.draw_pile, vec![CardKind::Copper; 4]);
         assert_eq!(p.hand, vec![CardKind::Copper]);
     }
 
```

Comment:
Suggest adding a test with an empty `draw_pile` and empty `discard_pile`. Should work now, right?